File: solutions/1637. Widest Vertical Area Between Two Points Containing No Points/1637.cpp
// Time complexity: O(sort)
// Space complexity: O(n)
class Solution {
 public:
  int maxWidthOfVerticalArea(vector<vector<int>>& points) {
    ranges::sort(points);

    int ans = 0;
    for (int i = 1; i < points.size(); ++i)
      ans = max(ans, points[i][0] - points[i - 1][0]);

    return ans;
  }
};

File: solutions/1637. Widest Vertical Area Between Two Points Containing No Points/1637.py
# Time complexity: O(sort)
# Space complexity: O(n)
class Solution:
  def maxWidthOfVerticalArea(self, points: List[List[int]]) -> int:
    points.sort(key=lambda v: v[0])
    return max(points[i+1][0] - points[i][0] for i in range(len(points) - 1))